### PR TITLE
fix: close file when errors occur

### PR DESCRIPTION
**Pull request by gunli**
## Proposed commit message
In filebeat/input/filestream/input.go, we forget to close the file when errors occur, this PR fix these bugs.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by pierrehilbert** [comment]
Hello @gunli 
Could you please sign our Contributor Agreement?

File: filebeat/input/filestream/input.go [hunk 1]
```diff
@@ -198,12 +198,13 @@ func (inp *filestream) open(log *logp.Logger, canceler input.Canceler, fs fileSo
 	// don't require 'complicated' logic.
 	logReader, err := newFileReader(log, canceler, f, inp.readerConfig, closerCfg)
 	if err != nil {
+		_ = f.Close()
```
[the remark this answers is not in the document]
Comment (rdner) [reply filebeat/input/filestream/input.go]: I made the change here https://github.com/elastic/beats/pull/37814/commits/c2211adc954daf4ba05d408c5bec9f1d5bd2471e

**Comment by rdner** [comment]
Although I fixed the PR I cannot merge it without a signed CLA, please sign the [CLA](https://www.elastic.co/contributor-agreement) @gunli

**Comment by rdner** [comment]
Closing in favour of https://github.com/elastic/beats/pull/37973

**Pull request closed without merging**